sktimokhina@edu.hse.ruPW1/sktimokhina@edu.hse.ruPW1/Presentation layer/Custom elements/Car.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  Car.swift
+//  sktimokhina@edu.hse.ruPW1
+//
+//  Created by Софья Тимохина on 16.09.2021.
+//
+
+import UIKit
+
+final class Car: UIView {
+    private var firstWheel: UIView!
+    private var secondWheel: UIView!
+    private var body: [UIView] = []
+    private var shouldAnimate: Bool = false
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        generateWheels()
+        generateWhelsBody()
+        generateBody()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func generateWheels() {
+        let wheelWidth = frame.width / 5
+        firstWheel = ShapeButton(viewModel:
```

Comment:
Предположительно кодстайл SwiftUI?

В свифте положено так:
```suggestion
                firstWheel = ShapeButton(
            viewModel: ShapeButton.ViewModel(
                height: wheelWidth,
                width: wheelWidth,
                x: wheelWidth,
                y: frame.height - wheelWidth,
                cornerRadius: wheelWidth / 1.6,
                backgroundColor: "#a5d6a7"
            )
        )
```